File: scripts/options/opt_a_commands.py
"""
OPTION A — Quick Commands (per-session button palette)
Adds GET/POST/DELETE routes to manage commands saved in commands.json.

Wire into routes.py:
    from opt_a_commands import register_command_routes
    register_command_routes(app)
"""
import json
import os
from flask import jsonify, request

COMMANDS_FILE = 'commands.json'


def _load():
    if not os.path.exists(COMMANDS_FILE):
        return {}
    with open(COMMANDS_FILE, 'r') as f:
        return json.load(f)


def _save(data):
    tmp = COMMANDS_FILE + '.tmp'
    with open(tmp, 'w') as f:
        json.dump(data, f, indent=2)
    os.replace(tmp, COMMANDS_FILE)


def register_command_routes(app):

    @app.route('/api/commands/<session>', methods=['GET'])
    def get_commands(session):
        return jsonify({'commands': _load().get(session, [])})

    @app.route('/api/commands/<session>', methods=['POST'])
    def add_command(session):
        data  = request.get_json() or {}
        label = data.get('label', '').strip()
        cmd   = data.get('command', '').strip()
        if not label or not cmd:
            return jsonify({'error': 'label and command required'}), 400
        store = _load()
        store.setdefault(session, []).append({'label': label, 'command': cmd})
        _save(store)
        return jsonify({'status': 'ok'})

    @app.route('/api/commands/<session>/<int:index>', methods=['DELETE'])
    def delete_command(session, index):
        store = _load()
        cmds  = store.get(session, [])
        if 0 <= index < len(cmds):
            cmds.pop(index)
            store[session] = cmds
            _save(store)
        return jsonify({'status': 'ok'})

File: scripts/options/opt_a_commands.js
// OPTION A — Quick Commands frontend
// Add a command palette below the toolbar. Buttons inject commands into the terminal.
//
// HTML needed:
//   <div id="quick-commands"></div>   ← place below #toolbar
//   <!-- Add-command form (optional) -->
//   <input id="cmd-label" placeholder="Label">
//   <input id="cmd-value" placeholder="Command">
//   <button onclick="addQuickCommand()">Add</button>

async function loadQuickCommands(session) {
  if (!session) {
    document.getElementById('quick-commands').innerHTML = '';
    return;
  }
  const { commands } = await fetch(
    `/api/commands/${encodeURIComponent(session)}`
  ).then(r => r.json());

  document.getElementById('quick-commands').innerHTML = commands.map((c, i) => `
    <button class="quick-cmd-btn"
            onclick="runQuickCommand(${JSON.stringify(c.command)})"
            title="${esc(c.command)}">
      ${esc(c.label)}
    </button>
    <button class="btn-kill" onclick="deleteQuickCommand(${i})" title="Remove">✕</button>
  `).join('');
}

function runQuickCommand(cmd) {
  if (!state.currentSession || !state.socket?.connected) return;
  // Exit copy-mode if active
  if (state.inCopyMode) {
    state.socket.emit('scroll', { session: state.currentSession, command: 'exit' });
    state.inCopyMode = false;
  }
  state.socket.emit('input', { session: state.currentSession, data: cmd + '\r' });
}

async function addQuickCommand() {
  const label = document.getElementById('cmd-label').value.trim();
  const cmd   = document.getElementById('cmd-value').value.trim();
  if (!label || !cmd || !state.currentSession) return;

  await fetch(`/api/commands/${encodeURIComponent(state.currentSession)}`, {
    method:  'POST',
    headers: { 'Content-Type': 'application/json' },
    body:    JSON.stringify({ label, command: cmd }),
  });
  document.getElementById('cmd-label').value = '';
  document.getElementById('cmd-value').value = '';
  loadQuickCommands(state.currentSession);
}

async function deleteQuickCommand(index) {
  await fetch(
    `/api/commands/${encodeURIComponent(state.currentSession)}/${index}`,
    { method: 'DELETE' }
  );
  loadQuickCommands(state.currentSession);
}

// Call loadQuickCommands(name) whenever the active session changes.
// Example: hook into activateSession():
//   const _origActivate = activateSession;
//   activateSession = name => { _origActivate(name); loadQuickCommands(name); };
